spyder/plugins/projects/tests/test_plugin.py

```diff
@@ -120,5 +156,49 @@ def test_open_project_uses_visible_config(projects, tmpdir, value):
     assert projects.dockwidget.isVisible() == value
 
 
+def test_set_get_project_filenames_when_closing(create_projects, tmpdir):
+    """
+    Test that the currently opened files in the Editor are saved and loaded
+    correctly to and from the project config when the project is closed and
+    then reopened.
+
+    Regression test for Issue #8375
+    """
+    opened_files = ['file1', 'file2', 'file3']
+    path = to_text_string(tmpdir.mkdir('project1'))
+
+    # Create the projects plugin.
+    projects = create_projects(path, opened_files)
+    assert projects.get_project_filenames() == []
```

Comment:
Why is this empty and not equal to `opened_files`?

Reply:
`get_project_filenames` returns what is saved in the project's config file and this gets updated only when closing the project, not as new files are opened or closed in the Editor.

Reply:
So `create_project` open a new project with an empty list of files at first in the config. Then when we close the project, the list of files we are mocking as open in the Editor are set as the project filenames in the config file.

Reply:
Ok, I see. Thanks for the explanation!